src/coprocessor/statistics/analyze.rs

```diff
@@ -220,7 +228,31 @@ impl<S: Snapshot> SampleBuilder<S> {
                     pk_builder.append(&v);
                 }
             }
-            for (collector, val) in collectors.iter_mut().zip(cols_iter) {
+            for (i, (collector, val)) in collectors.iter_mut().zip(cols_iter).enumerate() {
+                if self.cols_info[i].get_tp() == FieldTypeTp::TinyBlob as i32
```

Comment:
Maybe you can use `self.cols_info[i].as_accessor().is_string_like()`

Reply:
@winoros `as_accessor` is available for both `ColumnInfo` and `FieldType`, after `use xxxx::FieldTypeAccessor`